src/lib/components/event/event-group-details.svelte

```diff
@@ -7,10 +10,27 @@
     eventOrGroupIsTerminated,
   } from '$lib/models/event-groups/get-event-in-group';
   import { isLocalActivityMarkerEvent } from '$lib/utilities/is-event-type';
+  import { routeForEventHistory } from '$lib/utilities/route-for';
 
   export let eventGroup: EventGroup;
   export let selectedId: string;
   export let onGroupClick: (id: string) => void;
+
+  const handleGroupClick = (id: string) => {
+    const { namespace, workflow, run } = $page.params;
+    const queryParams: Record<string, string> = {};
+    $page.url.searchParams.forEach((value, key) => {
```

Comment:
Why not just use a `reduce`? This can probably be broken out into a utility function and unit tested. Now that I think about it, that's probably true for this entire function.